### PR TITLE
Feature/simple search

### PR DESCRIPTION
**Pull request by Arkitecht**
This adds a simple search feature (when enabled in config) to the main discussions page (which will search all discussion names and posts) and the discussion page (which will search all posts).

This is a simple search using a like search

### CODE REVIEW

**Comment by Arkitecht** [comment]
Sure, here are some screenshots:
Screen Shot 2022-10-13 at 8 16 40 AM
Screen Shot 2022-10-13 at 8 16 52 AM
Screen Shot 2022-10-13 at 8 17 10 AM
Screen Shot 2022-10-13 at 8 17 24 AM

File: resources/views/components/search.blade.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
+@props(['search' => ''])
+
+@if (\Firefly\Features::enabled('search'))
+    <form method="GET">
+        <div>
+            <label for="search" class="block text-lg font-medium text-gray-700 hidden">Quick search</label>
```
Comment (andreaselia) [inline resources/views/components/search.blade.php:6]: "block" and "hidden" on the same element? Do you mean to remove these and use "sr-only" or something?

File: resources/views/discussions/show.blade.php [hunk 2]
```diff
@@ -32,6 +32,8 @@
 </div>
 
 <div class="space-y-5">
+    <x-search :search="$search"></x-search>
```
Comment (andreaselia) [inline resources/views/discussions/show.blade.php:35]: Can we do this instead or am I wrong in assuming that this is a thing?
```suggestion
    <x-search :search="$search" />
```

File: resources/views/index.blade.php [hunk 3]
```diff
@@ -5,6 +5,8 @@
     {{ __('Discussions') }}
 </h2>
 
+<x-search :search="$search"></x-search>
```
Comment (andreaselia) [inline resources/views/index.blade.php:8]: Same as above here.

File: src/Models/Discussion.php [hunk 4]
```diff
@@ -143,4 +144,21 @@ public function scopeWithIsBeingWatched(Builder $builder, $user)
...
+                $query->where('title', 'like', '%'.$search.'%')
+                    ->orWhereExists(function ($query) use ($search) {
+                        $query->select(DB::raw(1))
+                            ->from('posts')
+                            ->whereColumn('discussion_id', 'discussions.id')
+                            ->where('content', 'like', '%'.$search.'%');
+                    });
+            });
+        });
+    }
```
Comment (andreaselia) [inline src/Models/Discussion.php:163]: @tomirons thoughts on this?

File: src/Models/Post.php [hunk 5]
```diff
@@ -49,4 +51,13 @@ public function getIsRichlyFormattedAttribute(): bool
...
     }
+
+    public function scopeWithSearch(Builder $builder, ?string $search)
+    {
+        $search = strtr($search, ['%' => '\%', '_' => '\_', '\\' => '\\\\']);
+
+        $builder->when(Features::enabled('search') && $search, function ($query) use ($search) {
+            $query->where('content', 'like', '%'.$search.'%');
+        });
+    }
```
Comment (andreaselia) [inline src/Models/Post.php:62]: @tomirons and this?

**Review by andreaselia — CHANGES_REQUESTED** [review]

File: resources/views/components/search.blade.php (hunk 1 above)
Comment (Arkitecht) [reply resources/views/components/search.blade.php:6]: yes, will update

File: resources/views/discussions/show.blade.php (hunk 2 above)
Comment (Arkitecht) [reply resources/views/discussions/show.blade.php:35]: yes, will update

**Comment by andreaselia** [comment]
@Arkitecht would you mind adding some tests for this functionality please?

**Comment by andreaselia** [comment]
You're awesome @Arkitecht, thanks for adding test coverage for this feature. Are you happy with the PR now? 🚀

**Comment by Arkitecht** [comment]
I am good with it if you guys are! Have a couple more updates coming soon :)

**Review by andreaselia — APPROVED** [review]

**Comment by andreaselia** [comment]
Just a few conflicts @Arkitecht from the previous PR, would you mind sorting them and then I can merge and tag a new release. I look forward to seeing the other updates coming soon, thank you!

**Comment by Arkitecht** [comment]
I have addressed all of the conflicts, and added the search to the group page as well, with test coverage.

Let me know if you see anything else off

**Comment by andreaselia** [comment]
Awesome, thanks @Arkitecht. It looks good to me so I'm going to merge it now, but won't tag a release until later as there are some general changes I may make to the package just before I tag the new release.

**Comment by andreaselia** [comment]
I ended up publishing v1.6.0 now as I'm not sure when I'll get time to complete some frontend updates.

**Comment by Arkitecht** [comment]
Awesome, thank you!

Teaser....
image

**Comment by andreaselia** [comment]
> Awesome, thank you!
> 
> Teaser.... image

Reactions 😍 I look forward to seeing your PR for it 😁

**Pull request merged**